File: src/food.c
#include <stdlib.h>

#include <raylib.h>
#include <raymath.h>

#include "collisions.h"
#include "constants.h"
#include "snake.h"

#include "food.h"

void InitializeNewFood(Vector2* food, Snake* snake) {
    do {
        food->x = rand() % NUM_CELLS_PER_SIDE;
        food->y = rand() % NUM_CELLS_PER_SIDE;
    } while(IsSnakeColliding(snake, *food));
}

void DrawFood(Vector2 food) {
    int x = (food.x) * CELL_SIZE;
    int y = (food.y) * CELL_SIZE;
    DrawRectangle(x, y, CELL_SIZE, CELL_SIZE, APPLE_COLOR);
}

File: SnakeGame/SnakeGame/Entities/Food.cs
using System.Numerics;
using Raylib_cs;
using SnakeGame.Game;

namespace SnakeGame.Entities;

public class Food
{
    public Vector2 Position { get; }
    public Color Color { get; } = new(211, 85, 86, 255);

    public Food(Snake snake)
    {
        var random = new Random();
        do
        {
            var x = random.Next(GameGrid.NumCellsPerSide);
            var y = random.Next(GameGrid.NumCellsPerSide); 
            Position = new Vector2(x, y);
        } while (snake.IsEating(this));
    }
    
    public void Draw()
    {
        var x = (int)Position.X * GameGrid.CellSize;
        var y = (int)Position.Y * GameGrid.CellSize;
        Raylib.DrawRectangle(x, y, GameGrid.CellSize, GameGrid.CellSize, Color);
    }
}